### PR TITLE
Make our code compliant to durable-task-plugin 1.18

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: CIScripts/Common/Invoke-NativeCommand.ps1 [hunk 1]
```diff
@@ -19,11 +19,11 @@ function Invoke-NativeCommand {
     #
     # Note: The command has to return 0 exitcode to be considered successful.
 
-    $Global:LastExitCode = $null
-
     & {
+        # Since we're redirecting stderr to stdout we shouldn't have to set ErrorActionPreference
+        # but because of a bug in Powershell we have to
```
Comment (m-kostrzewa) [inline CIScripts/Common/Invoke-NativeCommand.ps1]: Could you link to some existing bugs about this issue here?

**Comment by m-kostrzewa** [comment]
:clap:

File: CIScripts/Common/Invoke-NativeCommand.ps1 [hunk 2]
```diff
@@ -33,6 +33,4 @@ function Invoke-NativeCommand {
     if ($AllowNonZero) {
         Write-Output $LastExitCode
     }
-
-    $Global:LastExitCode = $null
```
Comment (krdln) [inline CIScripts/Common/Invoke-NativeCommand.ps1]: I guess this is not needed, because now the `$Error` [is checked before](https://github.com/jenkinsci/durable-task-plugin/blob/4a06f3cc8eb9053c0bf5597a06b69163c005c3b0/src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java#L103) `$LastExitCode`? Is my understanding correct?

Anyway, I'd keep this line, so we don't lose compatibility with currently used version of durable task plugin.
(also, it's needed, so next time `$LastExitCode -ne 0` fails on "Command not found" errors and the like)

Comment (mclapinski) [reply CIScripts/Common/Invoke-NativeCommand.ps1]: It passed CI, so it is compatible with current version of durable-task-plugin

Comment (krdln) [reply CIScripts/Common/Invoke-NativeCommand.ps1]: You're right, it passed CI, but unfortunatelly it detects less errors. With this line removed the following errors won't be catched:
1. Any error thrown after successful `Invoke-NativeCommand` (only with current version of durable task plugin)
2. Errors like "Command not found" (also with new version)

**Review by krdln — APPROVED** [review]
Thanks for adding comments with explanations of each hack!

MK (let's wait for pair review)

File: CIScripts/Common/Invoke-NativeCommand.ps1 [hunk 3]
```diff
@@ -1,7 +1,7 @@
 function Invoke-NativeCommand {
     Param (
-        [Parameter(Mandatory = $true)] [ScriptBlock] $ScriptBlock,
-        [Parameter(Mandatory = $false)] [Bool] $AllowNonZero = $false
+        [Parameter(Mandatory = $false)] [Bool] $AllowNonZero = $false,
```
Comment (krdln) [inline CIScripts/Common/Invoke-NativeCommand.ps1:3]: We can use flag here instead of parameter (didn't know about it when I've written it in the first place):
```
function Foo
{
  param(
    [switch] $AllowNonZero 
  )

  if ($AllowNonZero) { ... }
}
```

File: CIScripts/Build/BuildFunctions.ps1 [hunk 4]
```diff
@@ -270,12 +268,11 @@ function Run-Test {
...
     $Res = Invoke-Command -ScriptBlock {
-        $ErrorActionPreference = "SilentlyContinue"
-        $TestOutput = Invoke-NativeCommand -AllowNonZero $true -ScriptBlock {
+        $NativeCommandReturn = Invoke-NativeCommand -AllowNonZero $true -ScriptBlock {
             Invoke-Expression $TestExecutable
         }
-        $ExitCode = $TestOutput[-1]
-        $TestOutput.ForEach({ Write-Host $_ })
+        $ExitCode = $NativeCommandReturn[-1]
+        $TestOutput = $NativeCommandReturn[0..($NativeCommandReturn.Length-2)]
```
Comment (m-kostrzewa) [inline CIScripts/Build/BuildFunctions.ps1:275]: This isn't really clear, but it's not really part of this PR, so fix it if you want.

Maybe we should add:
1. `CaptureOutput` flag to `Invoke-NativeCommand`
2. Return a structure from `Invoke-NativeCommand` with two fields: `ExitCode` and `TestOutput`
3. If `CaptureOutput` is `$false`, then `TestOutput` would be `$null`

Comment (krdln) [reply CIScripts/Build/BuildFunctions.ps1:275]: > If CaptureOutput is $false, then TestOutput would be $null

And it should be live-streamed through `Write-Host`

**Review by m-kostrzewa — APPROVED** [review]

**Review by krdln — APPROVED** [review]
MK + MK

File: CIScripts/Build/BuildFunctions.ps1 (hunk 4 above)
Comment (mclapinski) [reply CIScripts/Build/BuildFunctions.ps1:275]: I tried to do this, it's actually pretty hard because PS never does what you expect it to do, so I won't do this.
